citadel/indico_citadel/search.py

```diff
@@ -0,0 +1,139 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2021 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+import re
+
+import requests
+from werkzeug.urls import url_join
+
+from indico.modules.search.base import IndicoSearchProvider, SearchTarget
+
+
+class CitadelProvider(IndicoSearchProvider):
+    def __init__(self, *args, **kwargs):
+        from indico_citadel.plugin import CitadelPlugin
+
+        super().__init__(*args, **kwargs)
+        self.token = CitadelPlugin.settings.get('search_backend_token')
+        self.backend_url = CitadelPlugin.settings.get('search_backend_url')
+        self.records_url = url_join(self.backend_url, 'api/records/')
+
+    def search(self, query, access, object_type=SearchTarget.event, page=1, params=None, highlight=True):
+        # https://cern-search.docs.cern.ch/usage/operations/#query-documents
+        # this token is used by the backend to authenticate and also to filter
+        # the objects that we can actually read
+        headers = {
+            'Authorization': f'Bearer {self.token}'
+        }
+
+        filter_query, ranges = format_filters(params)
+        # Look for objects matching the `query` and schema, make sure the query is properly escaped
+        # https://cern-search.docs.cern.ch/usage/operations/#advanced-queries
+        q = '{} {} +type:{}'.format(format_query(query), ranges, object_type.name)
```

Comment:
I think we could run `pyupgrade --py39-plus **/*.py` again; it will probably change this to an f-string.